x-pack/plugins/actions/server/builtin_action_types/xmatters.test.ts

```diff
@@ -0,0 +1,485 @@
+      headers: {
+        aheader: 'a value',
+      },
+      hasAuth: false,
+    };
+    const secrets: ActionTypeSecretsType = { user: null, password: null };
+    await actionType.executor({
+      actionId: 'some-id',
+      services,
+      config,
+      secrets,
+      params: {
+        alertActionGroup: 'small',
+        alertActionGroupName: 'Small t-shirt',
+        alertId: 'c9437cab-6a5b-45e8-bc8a-f4a8af440e97',
+        alertName: 'Test xMatters',
+        date: '2022-01-18T19:01:08.818Z',
+        severity: 'high',
+        spaceId: 'default',
+        tags: 'test1, test2',
+      },
+    });
+
+    expect(requestMock.mock.calls[0][0]).toMatchInlineSnapshot(`
+      Object {
+        "axios": undefined,
+        "configurationUtilities": Object {
+          "ensureActionTypeEnabled": [MockFunction],
+          "ensureHostnameAllowed": [MockFunction],
+          "ensureUriAllowed": [MockFunction],
+          "getCustomHostSettings": [MockFunction],
+          "getMicrosoftGraphApiUrl": [MockFunction],
+          "getProxySettings": [MockFunction],
+          "getResponseSettings": [MockFunction],
+          "getSSLSettings": [MockFunction],
+          "isActionTypeEnabled": [MockFunction],
+          "isHostnameAllowed": [MockFunction],
+          "isUriAllowed": [MockFunction],
+        },
+        "data": Object {
+          "alertActionGroup": "small",
+          "alertActionGroupName": "Small t-shirt",
+          "alertId": "c9437cab-6a5b-45e8-bc8a-f4a8af440e97",
+          "alertName": "Test xMatters",
+          "date": "2022-01-18T19:01:08.818Z",
+          "severity": "high",
+          "spaceId": "default",
+          "tags": "test1, test2",
+        },
+        "logger": Object {
+          "context": Array [],
+          "debug": [MockFunction] {
+            "calls": Array [
+              Array [
+                "response from xmatters action \\"some-id\\": [HTTP 200] ",
+              ],
+            ],
+            "results": Array [
+              Object {
+                "type": "return",
+                "value": undefined,
+              },
+            ],
+          },
+          "error": [MockFunction],
+          "fatal": [MockFunction],
+          "get": [MockFunction],
+          "info": [MockFunction],
+          "log": [MockFunction],
+          "trace": [MockFunction],
+          "warn": [MockFunction],
+        },
+        "method": "post",
+        "url": "https://abc.def/my-xmatters",
+      }
+    `);
+  });
+
+  test('renders parameter templates as expected', async () => {
+    const alertActionGroupName = `Small t-shirt`;
+    const alertId = `c9437cab-6a5b-45e8-bc8a-f4a8af440e97`;
+
+    // This line is failing at mand I am not sure why.
+    // The error says that renderParameterTemplates is undefined (I logged it and it's undefined, actionType is defined).
+    // This exact line (line 432) runs successfully in the webhook tests.
+    expect(actionType.renderParameterTemplates).toBeTruthy();
```

Comment:
I'm not sure why this line is having issues. Read comment above for more info. Will remove comment before taking this PR out of draft mode.